src/network/serverpackethandler.cpp

```diff
@@ -510,10 +510,6 @@ void Server::process_PlayerPos(RemotePlayer *player, PlayerSAO *playersao,
 	playersao->setWantedRange(wanted_range);
 
 	player->keyPressed = keyPressed;
-	player->control.up    = (keyPressed & (0x1 << 0));
-	player->control.down  = (keyPressed & (0x1 << 1));
-	player->control.left  = (keyPressed & (0x1 << 2));
-	player->control.right = (keyPressed & (0x1 << 3));
```

Comment:
These are still necessary to ensure API compatibility; for example `ObjectRef::l_get_player_control`.

Reply:
Is there a way I can check if a change is breaking the API?

Reply:
There's no general approach to this, but for this case you could test the PR using mods that make use of those four keys. Examples: boats, carts, mobs_redo + mob_horse

Reply:
boats, carts and horses work the same in latest stable 5.4.1 and this PR

![Screenshot_20210520_210012_minetest_joystick_horses](https://user-images.githubusercontent.com/9076163/119034463-a21a1b00-b9ae-11eb-89c3-fb11e7446b0f.png)

Reply:
This should not be a problem as this patch also changes the `l_get_player_control` API handler to include a backwards compatibility layer (for now): https://github.com/minetest/minetest/pull/10031/commits/afe491a7aac9c8f5dc60684be583ebb1018d73ac#diff-5cdd53776d4918652568f88b8e6a83b016edfffb0029a53e30cd3095f9e14f96